src/main/java/org/bananarama/crud/DeleteOperation.java

```diff
@@ -53,4 +53,11 @@
      * @return
      */
     public DeleteOperation<T> from(Stream<T> data,QueryOptions options);
+    
+    /**
+     * Delete all records from the persisted layer.
+     * @param data The collection of objects to be removed
+     * @return
+     */
+    public DeleteOperation<T> all();
```

Comment:
I suggest to declare this method as `default`, with a no-op implementation, maybe with a message logged. This would avoid all the conflicts with the existing adapters that do are not implementing the `all()` method.

Reference for [default methods](https://docs.oracle.com/javase/tutorial/java/IandI/defaultmethods.html)